src/main/java/util/SocketChannelUtil.java

```diff
@@ -0,0 +1,78 @@
+package util;
+
+import api.ByteBufferSerializableObject;
+import enums.PacketType;
+import impl.ByteBufferPacket;
+import impl.gen.PacketTypeStorage;
+import impl.gen.model.HeaderHolder;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.SocketChannel;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+//TODO - remove some serialize and deserialize methods, depends on what we need on current side (client/server).
+//TODO - add metadata to specification.
+
+public class SocketChannelUtil {
+
+    public SocketChannel getSocketChannel(String ip, int port) throws IOException {
+        SocketChannel channel = SocketChannel.open();
+        channel.connect(new InetSocketAddress(ip, port));
+
+        return channel;
+    }
+
+    public void writePacket(SocketChannel channel, ByteBufferPacket packet) throws IOException {
+        ByteBuffer byteBuffer = putObjectToByteBuffer(packet);
+
+        while(byteBuffer.hasRemaining()) {
+            channel.write(byteBuffer);
+        }
+    }
+
+    public ByteBufferSerializableObject readPacket(SocketChannel channel) throws Exception {
+        ByteBuffer completeBuffer = getByteBufferFromStream(channel, HeaderHolder.HEADER_SIZE);
+        HeaderHolder headerHolder = getHeaderFromByteBuffer(completeBuffer);
+        ByteBuffer bodyBuffer = getByteBufferFromStream(channel, headerHolder.size());
```

Comment:
This method can be called directly inside `getPacketBodyFromByteBuffer` if you change it signature to `getPacketBodyFromByteBuffer(SocketChannel channel, HeaderHolder headerHolder)`